lib/kubernetes-deploy/deploy_task.rb

```diff
@@ -568,7 +578,7 @@ def namespace_definition
 
     # make sure to never prune the ejson-keys secret
     def confirm_ejson_keys_not_prunable
-      secret = ejson_provisioner.ejson_keys_secret
+      secret = ejson_provisioners.first.ejson_keys_secret
```

Comment:
This is based on the assumption that `ejson_keys_secret` is always going to get `ejson-keys`. As above, this is currently going to be called once per template dir, but we can change that if so desired

Reply:
The avoidance of class-level state is what prompted me to choose not to do that in the first place. Even if we're able to create a stateless solution here, it opens the door/makes it easier for changes down the line that might violate that principle